### DISCUSSION TITLE
Binary version

### DISCUSSION THREAD

**Discussion by noxing** [e-m-b-a/emba#689] [Q&A]
I saw the binary version of the detection section when running the emba. The detection process seems to be complicated. There are actually binaries with no detected versions. How does it work? Thank you!

**Comment by m-1-k-3** [comment]
The binary version detection is handled from module [S09](https://github.com/e-m-b-a/emba/blob/master/modules/S09_firmware_base_version_check.sh). It iterates through the [configuration file](https://github.com/e-m-b-a/emba/blob/c9f15e31a66e6bd2561ce2ebf43f80eb2df07f75/modules/S09_firmware_base_version_check.sh#L38) and checks every binary in [bin_string_checker](https://github.com/e-m-b-a/emba/blob/c9f15e31a66e6bd2561ce2ebf43f80eb2df07f75/modules/S09_firmware_base_version_check.sh#L136C15-L136C15) with strings against the current identifier (see [here](https://github.com/e-m-b-a/emba/blob/c9f15e31a66e6bd2561ce2ebf43f80eb2df07f75/modules/S09_firmware_base_version_check.sh#L177))

There are multiple other cases included like the check of the extractor logs [here](https://github.com/e-m-b-a/emba/blob/c9f15e31a66e6bd2561ce2ebf43f80eb2df07f75/modules/S09_firmware_base_version_check.sh#L111) or some strict mode handling.

**Reply by noxing** [reply]
Thank you very much for your help.
